### PR TITLE
GEODE-9495: Limit Thread Sleep In PubSubNativeRedisAcceptanceTest

### PR DESCRIPTION
**Pull request by ringles**
The TIME_WAIT delay for Windows is much larger than either Linux or OSX. This patch allows the test class to determine the actual value of the 'socket reaping' delay in Linux and OSX so that the tests will complete much faster, particularly in CI.

### CODE REVIEW

File: geode-apis-compatible-with-redis/src/acceptanceTest/java/org/apache/geode/redis/internal/executor/pubsub/PubSubNativeRedisAcceptanceTest.java [hunk 1]
```diff
@@ -25,20 +31,50 @@
...
+    } else if (SystemUtils.IS_OS_MAC) {
+      try {
+        process = Runtime.getRuntime().exec("sysctl net.inet.tcp.msl");
+        BufferedReader reader = new BufferedReader(
+            new InputStreamReader(process.getInputStream()));
+        String line = reader.readLine();
+        String[] parts = line.split(":");
+        if (parts.length == 2) {
+          socketTimeWaitMsec = 2 * Long.parseLong(parts[1].trim());
+          System.out.println("TIME_WAIT: " + socketTimeWaitMsec);
```
Comment (dschneider-pivotal) [inline geode-apis-compatible-with-redis/src/acceptanceTest/java/org/apache/geode/redis/internal/executor/pubsub/PubSubNativeRedisAcceptanceTest.java:60]: remove the println?

**Review by dschneider-pivotal — CHANGES_REQUESTED** [review]

File: geode-apis-compatible-with-redis/src/acceptanceTest/java/org/apache/geode/redis/internal/executor/pubsub/PubSubNativeRedisAcceptanceTest.java (hunk 1 above)
Comment (ringles) [reply geode-apis-compatible-with-redis/src/acceptanceTest/java/org/apache/geode/redis/internal/executor/pubsub/PubSubNativeRedisAcceptanceTest.java:60]: Whoops, thought I'd done that!

**Review by jdeppe-pivotal — APPROVED** [review]

File: geode-apis-compatible-with-redis/src/acceptanceTest/java/org/apache/geode/redis/internal/executor/pubsub/PubSubNativeRedisAcceptanceTest.java [hunk 2]
```diff
@@ -25,20 +31,48 @@
...
+    }
+    // Just leave timeout at the default if it's some other OS or there's a problem getting OS value
+  }
+
+  private static String getCommandOutput(String commandString) throws IOException {
+    Process process = Runtime.getRuntime().exec(commandString);
+    BufferedReader reader = new BufferedReader(
+        new InputStreamReader(process.getInputStream()));
+    return reader.readLine();
+  }
```
Comment (kirklund) [inline geode-apis-compatible-with-redis/src/acceptanceTest/java/org/apache/geode/redis/internal/executor/pubsub/PubSubNativeRedisAcceptanceTest.java:65]: You should probably update to using `ProcessBuilder` and try-with-resources for the `InputStream`:
```java
Process process = new ProcessBuilder(commandString).start();
try (BufferedReader reader = new BufferedReader(
    new InputStreamReader(process.getInputStream()))) {
  return reader.readLine();
} finally {
  // probably not necessary but just to be sure test leaves no orphaned process
  process.destroy();
}
```

**Review by DonalEvans — APPROVED** [review]

File: geode-apis-compatible-with-redis/src/acceptanceTest/java/org/apache/geode/redis/internal/executor/pubsub/PubSubNativeRedisAcceptanceTest.java (hunk 2 above)
Comment (ringles) [reply geode-apis-compatible-with-redis/src/acceptanceTest/java/org/apache/geode/redis/internal/executor/pubsub/PubSubNativeRedisAcceptanceTest.java:65]: Doable, but I'd have to manually tokenize the command string - hand it a list, broken up where there's spaces. I suppose it's worth the extra effort, just to get the cleanup.

Comment (ringles) [reply geode-apis-compatible-with-redis/src/acceptanceTest/java/org/apache/geode/redis/internal/executor/pubsub/PubSubNativeRedisAcceptanceTest.java]: Done.

**Review by dschneider-pivotal — APPROVED** [review]

**Pull request merged**